### DISCUSSION TITLE
ModuleNotFoundError: No module named 'osmium.replication'

### DISCUSSION THREAD

**Discussion by Ash-47** [osmcode/pyosmium#228] [Q&A]
I've been wanting to update my local nominatim DB (for entire planet) and I've been following the steps mentioned in this article.
https://www.linuxbabe.com/ubuntu/osm-nominatim-geocoding-server-ubuntu-20-04
Basically I've been following step #6 & onwards
In step #6 after installing the osmium python module, and making changes to the local.php to point to the pyosmium-get-changes script, when I install the command $/data1/nominatim/build/utils/update.php --init-updates, I get the ModuleNotFoundError for osmium.replication.

Ubuntu version I'm on 20.04
osmium installed verison 3.6.0
I installed osmium via pip
I sudo switch user as postgres as it has access to the nominatim database & then run this:
$/srv/nominatim/build/utils/update.php --init-updates

The complete error I get:

![image](https://user-images.githubusercontent.com/63579562/220835473-4a1d31da-471a-4809-96b5-defbc5b35fd8.png)

content of my /nominatim/build/settings/local.php:

![image](https://user-images.githubusercontent.com/63579562/220836416-580aa0c4-4093-4a18-b98f-e38d400e76fa.png)

PS: In the script above the replication url is that of great-britain, but I eventually want it to be pointing to the the entire planet diffs, I suppose using https://planet.openstreetmap.org/replication/daily should suffice (Please advice if this is fine)

Thank you!

**Comment by lonvia** [comment]
Have you installed pyosmium as the user postgres? `pip install` will usually make the packages only available to the user that calls the command. So in theory you'd need to run `sudo -u postgres pip install osmium` but **don't do that**. postgres is a system user, you shouldn't run Nominatim stuff under that user. Create a separate one as described in the deployment docs.

**Reply by Ash-47** [reply]
Hey, I reinstalled osmium as postgres user as you had mentioned here, and post that the update command did work. I just wanted to know whether using https://planet.openstreetmap.org/replication/daily to run daily updates will be better or should I look for another source of data, but anyway, thank you for your inputs.

**Reply by lonvia** [reply]
So you completely ignored the part where I said **don't do that**.
